packages/ember-glimmer/lib/utils/references.js

```diff
@@ -127,6 +127,11 @@ export class PropertyReference extends CachedReference { // jshint ignore:line
   }
 
   get(propertyKey) {
+    if (typeof propertyKey === 'number') {
+      // Allows us to get numeric keys
+      propertyKey = '' + propertyKey;
```

Comment:
I kind of expect you would only need the logic in the keyword. What other use-cases does this cover?